sympy/printing/tests/test_fcode.py

```diff
@@ -75,7 +74,15 @@ def test_fcode_Float():
 def test_fcode_functions():
     x, y = symbols('x,y')
     assert fcode(sin(x) ** cos(y)) == "      sin(x)**cos(y)"
-    assert fcode(Max(x, y) + Min(x, y)) == "      max(x, y) + min(x, y)"
+
+
+def test_case():
+    x,x_,x__,y,X,X_,Y = symbols('x,x_,x__,y,X,X_,Y')
+    assert fcode(exp(x_) + sin(x*y) + cos(X*Y)) == \
+                        '      exp(x_) + sin(x*y) + cos(X*Y)'
+
+    assert fcode(exp(x__) + 2*x*Y*X_**Rational(7, 2)) == \
+                        '      2*X_**(7.0d0/2.0d0)*Y*x + exp(x__)'
```

Comment:
We should add tests for fcode(..., name_mangling=False) also. And some tests for an instance of FCodePrinter.